src/ProxyManager/ProxyGenerator/RemoteObject/MethodGenerator/RemoteObjectMethod.php

```diff
@@ -9,29 +9,76 @@
 use ReflectionClass;
 use Zend\Code\Generator\PropertyGenerator;
 use Zend\Code\Reflection\MethodReflection;
+use function count;
+use function strtr;
 use function var_export;
 
 /**
  * Method decorator for remote objects
  */
 class RemoteObjectMethod extends MethodGenerator
 {
+    private const TEMPLATE
+        = <<<'PHP'
+$defaultValues = #DEFAULT_VALUES#;
+$declaredParameterCount = #PARAMETER_COUNT#;
+
+$args = \func_get_args() + $defaultValues;
```

Comment:
Not a fan of `+` overall. Can `array_merge()` be used? What's the semantic difference here?

Reply:
Sadly array_merge would not overwrite the numeric indices, so had to go with `+`.